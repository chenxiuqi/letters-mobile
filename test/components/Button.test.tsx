import React from "react";
import { Button } from "@components";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

test("renders correctly", () => {
  const tree = renderer
    .create(
      <Button
        containerStyle={{ backgroundColor: "white" }}
        textStyle={{ fontSize: 10 }}
        buttonText="Button"
        onPress={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
