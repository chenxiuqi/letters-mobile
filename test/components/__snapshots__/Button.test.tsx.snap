// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 5,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "700",
          "padding": 6,
        },
        Object {},
        Object {
          "fontSize": 10,
        },
        Object {},
      ]
    }
  >
    Button
  </Text>
</View>
`;
